tests/map_find_test.cpp

```diff
@@ -107,20 +108,18 @@ BOOST_FIXTURE_TEST_CASE(should_not_copy_value_object_when_queried, MapFindFixtur
         const string str;
         NonCopyable(const string & str): str(str) {};
 
-        /// We remove the copy semantic to make sure objects are not duplicated along the way
-        NonCopyable(const NonCopyable &) = delete;
-        NonCopyable& operator=(const NonCopyable &) = delete;
-
         /// we only keep the move construction semantic to emplace into the map
         NonCopyable(NonCopyable && rhs): str(std::move(rhs.str)) {};
         NonCopyable& operator=(NonCopyable && rhs) = delete;
 
+    private:
+        BOOST_MOVABLE_BUT_NOT_COPYABLE(NonCopyable)
     };
 
     std::map<int, NonCopyable> non_cop_map;
-    non_cop_map.emplace( 1, string("one"));
-    non_cop_map.emplace( 2, string("two"));
-    non_cop_map.emplace( 3, string("three"));
+    non_cop_map.insert( std::make_pair(1, NonCopyable("one")) );
+    non_cop_map.insert( std::make_pair(2, NonCopyable("two")) );
+    non_cop_map.insert( std::make_pair(3, NonCopyable("three")) );
```

Comment:
I think now it's :
NonCopyable ------(move)--------> std::pair --------(copy)------->std::map

@TeXitoi what do you think ?

Reply:
okay my mistake

Reply:
If you have some fun, you can have a wee look at std::map's implementation: https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/stl_map.h#L815 :laughing:

Reply:
:smiley: